dubbo-rpc/dubbo-rpc-triple/src/main/java/org/apache/dubbo/rpc/protocol/tri/h12/grpc/GrpcRequestHandlerMapping.java

```diff
@@ -42,9 +43,16 @@ protected boolean supportContentType(String contentType) {
 
     @Override
     protected void determineHttpMessageCodec(RequestHandler handler, URL url, HttpRequest request) {
-        HttpMessageCodec codec = CODEC_FACTORY.createCodec(url, getFrameworkModel(), request.contentType());
-        handler.setHttpMessageDecoder(codec);
-        handler.setHttpMessageEncoder(codec);
+        GrpcCompositeCodec grpcCompositeCodec =
+                (GrpcCompositeCodec) CODEC_FACTORY.createCodec(url, getFrameworkModel(), request.contentType());
+        MethodDescriptor methodDescriptor = DescriptorUtils.findMethodDescriptor(
+                handler.getServiceDescriptor(), handler.getMethodName(), handler.isHasStub());
+        if (methodDescriptor != null) {
+            handler.setMethodDescriptor(methodDescriptor);
+            grpcCompositeCodec.loadPackableMethod(methodDescriptor);
+        }
+        handler.setHttpMessageDecoder(grpcCompositeCodec);
+        handler.setHttpMessageEncoder(grpcCompositeCodec);
```

Comment:
**Security Issue**: The method determineHttpMessageCodec dynamically loads a codec based on the request content type without validating the content type, which could lead to loading of unexpected or malicious codecs. <br> **Fix**: Implement validation for the content type against a whitelist of allowed content types before loading the codec. <br> **Code Suggestion**: 
 ```
 protected void determineHttpMessageCodec(RequestHandler handler, URL url, HttpRequest request) {
     if (!isValidContentType(request.contentType())) {
         throw new IllegalArgumentException("Invalid content type");
     }
     GrpcCompositeCodec grpcCompositeCodec =
             (GrpcCompositeCodec) CODEC_FACTORY.createCodec(url, getFrameworkModel(), request.contentType());
     MethodDescriptor methodDescriptor = DescriptorUtils.findMethodDescriptor(
             handler.getServiceDescriptor(), handler.getMethodName(), handler.isHasStub());
     if (methodDescriptor != null) {
         handler.setMethodDescriptor(methodDescriptor);
         grpcCompositeCodec.loadPackableMethod(methodDescriptor);
     }
     handler.setHttpMessageDecoder(grpcCompositeCodec);
     handler.setHttpMessageEncoder(grpcCompositeCodec);
 }

 private boolean isValidContentType(String contentType) {
     // Implementation of content type validation against a whitelist
     return true; // Simplified for this example
 }
 ```